### DISCUSSION TITLE
Should magit use `diff.interHunkContext` instead of `diff.context`?

### DISCUSSION THREAD

**Discussion by hab25** [magit/magit#5147] [Support] [answered]
I.e., in https://github.com/magit/magit/blob/f9268a959828d0c6ab26171dd2fb1ffc55e5ae70/lisp/magit-diff.el#L1476 

IIUC this is more appropriate because magit fuses hunks based on the value retrieved from that line of code.

If git's documentation about `interHunkContext` (https://git-scm.com/docs/git-diff#Documentation/git-diff.txt-diffinterHunkContext) isn't clear, this stackoverflow Q&A could help: https://stackoverflow.com/questions/40419193/git-how-to-configure-the-hunks-recognition .

**Comment by tarsius** [comment]
> Should magit use `diff.interHunkContext` instead of `diff.context`?

This feature was added to do the thing that it does do, so replacing it with another feature wouldn't be the right move. Maybe Magit should *also* provide a convenient way to change the value of the other variable.

> IIUC this is more appropriate because magit fuses hunks based on the value retrieved from that line of code.

I don't think doing the other thing would be more (or less) "appropriate", it's just another, though related, feature.

Magit itself doesn't fuse hunks. Magit calls `git diff` which then does its thing based on arguments and variables, and Magit presents the output to the user. It does not do any "fusing of hunks" by itself.

For both variables I doubt that many users frequently change the value. The existing feature might be a bit overkill. I don't plan to duplicate it for the other feature.

If you have a preferred value, which is different from the default, I think your best option is to just once set this variable globally.

If you want to change the value frequently, you can add the argument yourself. Modify a copy of `magit-diff:-U` and also see https://magit.vc/manual/transient/Modifying-Existing-Transients.html.

**Reply by hab25** [reply, marked as answer]
My incorrect statements were related to me thinking that magit did the fusing itself; which in turn is related to a problem in my configuration; thanks for clarifying!
